src/order/dto/order.dto.ts

```diff
@@ -0,0 +1,26 @@
+import { IsArray, IsEnum, IsNotEmpty, IsNumber } from 'class-validator';
+import { OrderStatusEnum } from '../types/order-status-enum';
+import { PaymentStatusEnum } from '../types/payments-types-enum';
+
+export class CreateOrderDto {
+  @IsEnum(OrderStatusEnum, {
+    message: `$property must be one of these: ${Object.keys(OrderStatusEnum)}`,
+  })
+  status: OrderStatusEnum;
+  @IsNumber()
+  @IsNotEmpty()
+  price: number;
+  payment: {
+    cardholder: string;
+    paymentStatus: PaymentStatusEnum;
+    value: number;
+    dueDate: Date;
```

Comment:
I think that create a new DTO to implements this field 
```suggestion
  payment: PaymentDto
```
Please, put the correct decorators for each field
```
export class PaymentDto {
    cardholder: string;
    paymentStatus: PaymentStatusEnum;
    value: number;
    dueDate: Date;
}
```